gbbs/benchmarks/BatchDynamicConnectivity/EulerTourTree/ETTree.h

```diff
@@ -591,6 +591,38 @@ struct ETTree {
         return edge_sum;
         //return edge_sum.first ^ edge_sum.second;
     }
+    
+    template <class KY, class VL, class HH>
+    sequence<sequence<sequence<std::pair<uintE, uintE>>>> batch_sub_tree_sum(sequence<std::pair<uintE,uintE>> v_p,
+            pbbslib::sparse_table<KY, VL, HH> edge_index_table)
+    {
+
+
+        size_t num_queries = v_p.size();
+  
+        sequence<sequence<sequence<std::pair<uintE, uintE>>>> subtree_sums(num_queries);
+
+        parallel_for(0, num_queries, [&](size_t i){
+
+            uintE u = v_p[i].first; 
+            uintE v = v_p[i].second; 
+            auto index_uv = edge_index_table.find(std::make_pair(u, v), UINT_E_MAX);
+            if (index_uv == UINT_E_MAX)
+            std::cout << "This is an error in edge_index_table in get_subtree_sum" << std::endl;
```

Comment:
```suggestion
            std::cerr << "This is an error in edge_index_table in get_subtree_sum" << std::endl;
```
Error messages should go to the standard error stream not the standard output stream. (Consider also using exceptions, depending on whether they are common in this code base and whether the parlay random for needs a `nothrow` lambda.)

Reply:
For this part, since the original code of ETTree.h is using std::cout, I chose to follow this pattern. Besides, it is true that I am not aware that std:;cerr is an option :) 

 I am not very familiar about the error stream and exception mechanism for c++, I may have to read about it a little  bit first before making changes to the code. ( Since I assume it would be better to alter the code of handing errors in all parts of this source file to maintain consistency?

Reply:
Good point! For this code, you may want to maintain backwards compatibility with style.

[This StackOverflow post](https://stackoverflow.com/a/60558096/2769271) gives all the detail you really need to understand `cout` vs `cerr`. When we run experiments on servers like Compute Canada, it will be useful to be able to redirect normal output and errors into different files so that it is easier to write python scripts to automatically turn the normal program output into plots for papers and presentations.